cli/azd/cmd/env.go

```diff
@@ -448,8 +481,8 @@ func newEnvGetValuesAction(
 	formatter output.Formatter,
 	writer io.Writer,
 	azCli azcli.AzCli,
-	flags envGetValuesFlags,
-) *envGetValuesAction {
+	flags *envGetValuesFlags,
+) actions.Action {
```

Comment:
Is it possible to continue to return the concrete type instead of the interface type here?  It's slightly more idiomatic to return concrete types rather than interfaces values, but I think perhaps because of how we do the registration with the IoC container, this has to be the interface type?

Reply:
At this time no.  The actions are registered as named registrations of type `actions.Action` within the container because at resolution time we don't have the generic type safety we need to resolve it to the correct type.

I hope that Go enhances their generic support to have generic function type parameters in addition to struct type parameters.  Once we have that we should be able to do something like this.